docs/integrations/builtin/cluster-nodes/root-nodes/n8n-nodes-langchain.agent/tools-agent.md

```diff
@@ -64,6 +97,45 @@ Refine the Tools Agent node's behavior using these options:
 
 Refer to the main AI Agent node's [Templates and examples](/integrations/builtin/cluster-nodes/root-nodes/n8n-nodes-langchain.agent/index/#templates-and-examples) section.
 
+## Dynamic parameters for tools
+
+When configuring tools connected to the Tools Agent, you can use the `$fromAI()` function to dynamically populate parameter values using the AI model. The AI model will fill in appropriate data given the context from the task and information from other connected tools.
+
+/// note | Only for the Tools Agent
+The `$fromAI()` function is only available for tools connected to the Tools Agent. You also can't use the `$fromAI()` function with the [Call n8n Workflow](/integrations/builtin/cluster-nodes/sub-nodes/n8n-nodes-langchain.toolworkflow/) tool.
```

Comment:
Just a minor clarification — for now it's only available to Node Tools. So you can't use it in "Call n8n Workflow Tool", "Code Tool" or "HTTP Request Tool". Those have different methods to populate the parameters.

Reply:
Ah, thanks for the clarification. What exactly do we mean by "Node Tools"? I'm having trouble finding that in our docs, so I want to understand if that's a term I should know.

Reply:
That would be only the wrapped App nodes, so:
* [Airtable](/integrations/builtin/app-nodes/n8n-nodes-base.airtable/)
* [Baserow](/integrations/builtin/app-nodes/n8n-nodes-base.baserow/)
* [Gmail](/integrations/builtin/app-nodes/n8n-nodes-base.gmail/)
* [Google Calendar](/integrations/builtin/app-nodes/n8n-nodes-base.googlecalendar/)
* [Google Docs](/integrations/builtin/app-nodes/n8n-nodes-base.googledocs/)
* [Google Drive](/integrations/builtin/app-nodes/n8n-nodes-base.googledrive/)
* [Google Sheets](/integrations/builtin/app-nodes/n8n-nodes-base.googlesheets/)
* [Hacker News](/integrations/builtin/app-nodes/n8n-nodes-base.hackernews/)
* [Jira Software](/integrations/builtin/app-nodes/n8n-nodes-base.jira/)
* [Microsoft Outlook](/integrations/builtin/app-nodes/n8n-nodes-base.microsoftoutlook/)
* [Microsoft SQL](/integrations/builtin/app-nodes/n8n-nodes-base.microsoftsql/)
* [MongoDB](/integrations/builtin/app-nodes/n8n-nodes-base.mongodb/)
* [MySQL](/integrations/builtin/app-nodes/n8n-nodes-base.mysql/)
* [NocoDB](/integrations/builtin/app-nodes/n8n-nodes-base.nocodb/)
* [Notion](/integrations/builtin/app-nodes/n8n-nodes-base.notion/)
* [Postgres](/integrations/builtin/app-nodes/n8n-nodes-base.postgres/)
* [Redis](/integrations/builtin/app-nodes/n8n-nodes-base.redis/)
* [Send Email](/integrations/builtin/core-nodes/n8n-nodes-base.sendemail/)
* [Slack](/integrations/builtin/app-nodes/n8n-nodes-base.slack/)
* [Supabase](/integrations/builtin/app-nodes/n8n-nodes-base.supabase/)
* [Telegram](/integrations/builtin/app-nodes/n8n-nodes-base.telegram/)
* [WooCommerce](/integrations/builtin/app-nodes/n8n-nodes-base.woocommerce/)